src/test/java/org/mskcc/cbio/oncokb/web/rest/SlackControllerIT.java

```diff
@@ -0,0 +1,647 @@
+        values.put(BlockId.SUBJECT_INPUT.getId(), subjectMap);
+        values.put(BlockId.BODY_INPUT.getId(), bodyMap);
+        viewState.setValues(values);
+        view.setState(viewState);
+        viewJSON.setView(view);
+
+        return viewJSON;
+    }
+
+    private void checkSlackBlock(String url, Payload payload, ActionId actionId, boolean collapsed) {
+        assertThat(url).isEqualTo(USER_REGISTRATION_WEBHOOK);
+
+        // Add required values
+        Set<String> expectedValues = new HashSet<>();
+        expectedValues.add(DEFAULT_USER_EMAIL);
+        expectedValues.add(DEFAULT_COMPANY_NAME);
+        if (collapsed) {
+            expectedValues.add(BlockId.COLLAPSED.getId());
+            expectedValues.add(actionId == ActionId.CHANGE_LICENSE_TYPE ? OTHER_LICENSE_TYPE.getShortName() : DEFAULT_LICENSE_TYPE.getShortName());
+            if (actionId == ActionId.GIVE_TRIAL_ACCESS) {
+                expectedValues.add("TRIAL");
+            } else if (ActionId.isConfirmEmailAction(actionId)) {
+                for (DropdownEmailOption option : DropdownEmailOption.values()) {
+                    if (option.getActionId() == actionId && option.getCollapsedNote().isPresent()) {
+                        expectedValues.add(option.getCollapsedNote().get());
+                    }
+                }
+            }
+        } else {
+            expectedValues.add(actionId == ActionId.CHANGE_LICENSE_TYPE ? OTHER_LICENSE_TYPE.getName() : DEFAULT_LICENSE_TYPE.getName());
+            if (ActionId.isConfirmEmailAction(actionId)) {
+                for (DropdownEmailOption option : DropdownEmailOption.values()) {
+                    if (option.getConfirmActionId() == Optional.of(actionId)) {
+                        expectedValues.add(option.getExpandedNote());
+                    }
+                }
+            } else if (actionId == ActionId.APPROVE_USER) {
+                expectedValues.add(SlackService.APPROVE_USER_EXPANDED_NOTE);
+            } else if (actionId == ActionId.CONVERT_TO_REGULAR_ACCOUNT) {
+                expectedValues.add(SlackService.CONVERT_TO_REGULAR_ACCOUNT_EXPANDED_NOTE);
+            }
+        }
+
+        // Search for required values
+        for (LayoutBlock block : payload.getBlocks()) {
+            if (block instanceof SectionBlock) {
+                SectionBlock sectionBlock = (SectionBlock) block;
+                if (sectionBlock.getText() != null) {
+                    expectedValues.removeIf(sectionBlock.getText().getText()::contains);
+                }
+                if (sectionBlock.getFields() != null) {
+                    for (TextObject textObject : sectionBlock.getFields()) {
+                        expectedValues.removeIf(textObject.getText()::contains);
+                    }
+                }
+                if (sectionBlock.getBlockId() != null) {
+                    expectedValues.removeIf(sectionBlock.getBlockId()::contains);
+                }
+            }
+        }
```

Comment:
I see few places using this strategy. Is this the only option to test? What happens when we have same value in different positions within the block?

Reply:
I wanted to make the test as general as possible, so just check every location of every block. Because we may change the format of the Slack Integration based on preferences that come up. This just ensures all the crucial information is at least somewhere in the blocks. If a value appears twice, it will pass on the first one and ignore the second. Thoughts?

Reply:
Search by BlockId (create issue)